### PR TITLE
Feature graphql backend integration

### PR DESCRIPTION
**Pull request by NotSugat**

### CODE REVIEW

**Comment by abhiyandhakal** [comment]
image
- why tasks?
- Don't you think description box should be bigger? like a textarea?
- priority **must** be a dropdown.

**Comment by abhiyandhakal** [comment]
Data is being fetched everytime... better have all of them already in the state... feels really slow...

**Comment by abhiyandhakal** [comment]
image
Date could be improved

**Comment by abhiyandhakal** [comment]
Do you need `taskId` from the backend in `Todo` type?
image

**Comment by abhiyandhakal** [comment]
image
clicking on the checkbox throws this error:
image

**Comment by abhiyandhakal** [comment]
image
Delete doesn't work

**Comment by abhiyandhakal** [comment]
remove these:
image

**Comment by NotSugat** [comment]
> Do you need `taskId` from the backend in `Todo` type?
> image
> 

Yes

File: src/app/components/goals/CreateGoalDialog.tsx
[the remark this answers is not in the document]
Comment (NotSugat) [reply src/app/components/goals/CreateGoalDialog.tsx:1]: create Goal task dialog will change after adding goal questions as well so just implemented graphql and added pull request and ui changes will be done in another branch.

**Pull request merged**